common/cuda_hip/base/types.hpp

```diff
@@ -8,8 +8,12 @@
 
 #if defined(GKO_COMPILING_CUDA)
 #include "cuda/base/types.hpp"
+#define device_type cuda_type
+#define as_device_type as_cuda_type
```

Comment:
aren't those already defined in the `cuda/base/types.hpp` header? Same for hip.

Reply:
ah yeah, missed those. Those should not be macros, as that breaks other files